internal/runner/service_test.go

```diff
@@ -218,6 +218,61 @@ func Test_runnerService(t *testing.T) {
 		assert.EqualValues(t, 0, result.Responses[1].ExitCode.Value)
 	})
 
+	// The longest accepted line must not have more than 1024 bytes,
+	// including the new line character at the end. Any line longer results in ^G (BELL).
+	// It is possible to send more data, but it must be divided in 1024-byte chunks
+	// separated by the new line character (\n).
+	// TODO(adamb): sort out the root cause of this limitation.
```

Comment:
@mxsdev @sourishkrout this is a quite strange limitation I found on my local machine. On Linux, I can write 4096. This limit can be found on [the termios man page](https://man7.org/linux/man-pages/man3/termios.3.html):

> The maximum line length is 4096 chars (including the
         terminating newline character); lines longer than 4096 chars
         are truncated.  After 4095 characters, input processing (e.g.,
         ISIG and ECHO* processing) continues, but any input data after
         4095 characters up to (but not including) any terminating
         newline is discarded.  This ensures that the terminal can
         always receive more input until at least one line can be read.

I haven't found yet which settings is responsible for lower limit on macOS 12.6.

Reply:
I found [it](https://github.com/apple/darwin-xnu/blob/main/bsd/sys/syslimits.h#L98)! It's mentioned in [the manual](https://man.freebsd.org/cgi/man.cgi?query=termios&sektion=4) as `{MAX_INPUT}`.